---
title: "R_for_bioinformatics_2"
author: "HibaBenAribi"
date: "2024-04-16"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# dependencies
```{r }
#install.packages("BiocManager")
#BiocManager::install("clusterProfiler")
#BiocManager::install("enrichplot")
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
```
# Gene expression analysis
```{r }
# read data
expression <- read.csv("source data/expression-data.csv")
# define filter parameter
deg_pval <- 0.05 # max p value to define DEGs
deg_logFC <- 2 # min log fc value 
expression$diffexpressed <- "Not Significant"
expression$diffexpressed[expression$log2FoldChange  > deg_logFC  & expression$pvalue < deg_pval ] <- "Upregulated genes"
expression$diffexpressed[expression$log2FoldChange  < -deg_logFC & expression$pvalue < deg_pval] <- "Downregulated genes"

```
#filter DEG

```{r }
DEGs <- as.data.frame(expression[expression$diffexpressed != "Not Significant",])

```
# ontology analysis
```{r }
ontology <- enrichGO(gene = DEGs$gene, OrgDb="org.Hs.eg.db", keyType="SYMBOL", ont='ALL')

```

# visualize ontology analysis data
```{r }
# bar plot
 barplot(ontology, showCategory=10)

```


```{r }
# Dotplot
dotplot(ontology, showCategory=10)

```