---
title: "Bioinformatics_with_R_1"
author: "HibaBenAribi"
date: "2024-04-16"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# call dependencies
```{r }
library(readr)
library(dplyr)
library(ggplot2)
library(readxl)
```
# read expression data file
## expression data are RNA sequencing data
```{r }
# import data
expression <- read.csv("source data/expression-data.csv")
```
# determine Differentially expressed genes (DEGs)
## p value to determine significant changes between 2 groups (control vs patients)
## log FC to define if the genes are upregulated or downregulated


```{r }
# define filter parameter
deg_pval <- 0.05 # max p value to define DEGs
deg_logFC <- 2 # min log fc value 

```

# if logFC is positive the genes are upregulated
# upregulated gene are expressed in patient more than controls

# if logFC is negative the genes are downregulated
# downregulated gene are expressed in  controls more than  patient
```{r }
expression$diffexpressed <- "Not Significant"
expression$diffexpressed[expression$log2FoldChange  > deg_logFC  & expression$pvalue < deg_pval ] <- "Upregulated genes"
expression$diffexpressed[expression$log2FoldChange  < -deg_logFC & expression$pvalue < deg_pval] <- "Downregulated genes"
```

#see new column
```{r }
# Count the number of each type
gene_counts <- table(expression$diffexpressed)

# Print the counts
print(gene_counts)
```
# Visualize the results as a volcano plot
```{r }
ggplot(data=expression, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed)) +
  geom_point() + theme_minimal() + theme(text = element_text(size = 15))
```